lib/util/git.js

```diff
@@ -113,57 +131,51 @@ function revs (repo, opts) {
   return pinflight(`ls-remote:${repo}`, () => {
     return spawnGit(['ls-remote', '-h', '-t', repo], {
       env: gitEnv()
-    }, opts).then(child => {
-      let stdout = ''
-      let stderr = ''
-      child.stdout.on('data', d => { stdout += d })
-      child.stderr.on('data', d => { stderr += d })
-      return finished(child).catch(err => {
-        err.message = `Error while executing:\n${GITPATH} ls-remote -h -t ${repo}\n\n${stderr}\n${err.message}`
-        throw err
-      }).then(() => {
-        return stdout.split('\n').reduce((revs, line) => {
-          const split = line.split(/\s+/, 2)
-          if (split.length < 2) { return revs }
-          const sha = split[0].trim()
-          const ref = split[1].trim().match(/(?:refs\/[^/]+\/)?(.*)/)[1]
-          if (!ref) { return revs } // ???
-          if (ref.endsWith(CARET_BRACES)) { return revs } // refs/tags/x^{} crap
-          const type = refType(line)
-          const doc = {sha, ref, type}
-
-          revs.refs[ref] = doc
-          // We can check out shallow clones on specific SHAs if we have a ref
-          if (revs.shas[sha]) {
-            revs.shas[sha].push(ref)
-          } else {
-            revs.shas[sha] = [ref]
-          }
+    }, opts).then((stdout) => {
+      return stdout.split('\n').reduce((revs, line) => {
+        const split = line.split(/\s+/, 2)
+        if (split.length < 2) { return revs }
+        const sha = split[0].trim()
+        const ref = split[1].trim().match(/(?:refs\/[^/]+\/)?(.*)/)[1]
+        if (!ref) { return revs } // ???
+        if (ref.endsWith(CARET_BRACES)) { return revs } // refs/tags/x^{} crap
+        const type = refType(line)
+        const doc = {sha, ref, type}
 
-          if (type === 'tag') {
-            const match = ref.match(/v?(\d+\.\d+\.\d+)$/)
-            if (match) {
-              revs.versions[match[1]] = doc
-            }
-          }
+        revs.refs[ref] = doc
+        // We can check out shallow clones on specific SHAs if we have a ref
+        if (revs.shas[sha]) {
+          revs.shas[sha].push(ref)
+        } else {
+          revs.shas[sha] = [ref]
+        }
 
-          return revs
-        }, {versions: {}, 'dist-tags': {}, refs: {}, shas: {}})
-      }).then(revs => {
-        if (revs.refs.HEAD) {
-          const HEAD = revs.refs.HEAD
-          Object.keys(revs.versions).forEach(v => {
-            if (v.sha === HEAD.sha) {
-              revs['dist-tags'].HEAD = v
-              if (!revs.refs.latest) {
-                revs['dist-tags'].latest = revs.refs.HEAD
-              }
-            }
-          })
+        if (type === 'tag') {
+          const match = ref.match(/v?(\d+\.\d+\.\d+)$/)
```

Comment:
This line is conflicting with the version in `latest` -- could you pull in that patch here?

Reply:
done